test/jdk/javax/swing/JButton/HtmlButtonWithTextAndIcon.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+/* @test
+ * @bug 8302173
+ * @requires (os.family == "mac")
```

Comment:
Is this required? HtmlButtonImageTest is being tested for all platforms even though fix was in mac so I think we should make this test also platform-agnostic..

Reply:
I would but different L&F's show HTML content differently. For example, here's Metal:
![failImage](https://user-images.githubusercontent.com/96267980/220711014-a9b6be59-9a53-4044-9763-fffe157c5d73.png)

Other L&F's are slightly different as well, and this would require the test to handle each L&F differently. The changes were only for Aqua L&F, and the original issue fix was also for only Aqua L&F. So, I made this test only for macOS && Aqua L&F.